scripts/check-package.js

```diff
@@ -0,0 +1,173 @@
+        level: 'warn',
+        message: `${pkgName} should be at version ${versionOpt} (it is at ${pkgVersion})`
+      }
+    }
+  }
+
+/* Ensures a dep is installed with another dep */
+const hasOtherDep = (pkgName, depKey, message) =>
+  function*({ pkgName, depKey, pkgJson }) {
+    if (!pkgJson[depKey] || !pkgJson[depKey][pkgName]) {
+      yield {
+        level: 'warn',
+        message
+      }
+    }
+  }
+
+const isStrict = version =>
+  Boolean(/^[0-9]/.exec(version)) && version.split('.').length === 3
+
+const peerDepHasLenientVersion = () =>
+  function*({ depKey, pkgName, pkgVersion }) {
+    if (depKey !== 'peerDependencies') {
+      return
+    }
+    if (isStrict(pkgVersion)) {
+      yield {
+        level: 'warn',
+        message: `Incorrect version ${pkgName}@${pkgVersion} : peer deps should have lenient version`
+      }
+    }
+  }
+
+const constraints = [
+  [isBabelDevPackage, [uniqueVersion(), depKeyIs('devDependencies')]],
+  [isPkg('react-markdown'), [shouldNotBeUsed()]],
+  [isPkg('@babel/runtime'), [uniqueVersion(), depKeyIs('peerDependencies')]],
+  [
+    isPkg('jest'),
+    [
+      uniqueVersion(),
+      versionAt('26.2.2'),
+      depKeyIs('devDependencies'),
+      hasOtherDep(
+        'babel-jest',
+        'devDependencies',
+        'babel-jest must be installed along jest'
+      )
+    ]
+  ],
+  [any, [peerDepHasLenientVersion()]],
+  [isPkg('cozy-client'), [depKeyIs(['peerDependencies', 'devDependencies'])]],
+  [isPkg('cozy-ui'), [depKeyIs(['peerDependencies', 'devDependencies'])]],
+  [isPkg('cozy-realtime'), [depKeyIs(['peerDependencies', 'devDependencies'])]],
+  [isPkg('babel-core'), [shouldNotBeUsed()]]
+]
+
+/* Reads all the package.json and log warnings when constraints are not met */
+const main = async () => {
+  const packages = fs.readdirSync('packages')
+  for (let pkgName of packages) {
+    const pkgJson = readJsonSync(path.join('packages', pkgName, 'package.json'))
+    if (!pkgJson.devDependencies) {
+      continue
+    }
+    if (pkgJson.private) {
+      continue
+    }
+    for (let depKey of [
+      'dependencies',
+      'devDependencies',
+      'peerDependencies'
+    ]) {
+      if (!pkgJson[depKey]) {
+        continue
+      }
+      const deps = Object.keys(pkgJson[depKey])
+      for (let dep of deps) {
+        for (let [pred, checks] of constraints) {
+          if (!pred(dep)) {
+            continue
+          }
+          for (let c of checks) {
+            const results = Array.from(
+              c({
+                pkgName: dep,
+                pkgVersion: pkgJson[depKey][dep],
+                depKey,
+                pkgJson
+              })
+            )
+            for (let result of results) {
+              console.log(`[${result.level}] ${pkgName}: ${result.message}`)
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+if (require.main === module) {
+  main().catch(e => {
+    console.error(e)
+    process.exit(1)
+  })
+}
```

Comment:
Wondering, should this script be in cozy-repo-doctor? Some functions can be very specific to cozy-lib's mono repo, but checking if the peer is met or else can be pretty generic

Reply:
I also wondered about that, I think it could be a good idea. Would you be up for it ? :D

Reply:
I've added an issue because it is not so clear now to me what would be the advantages and how we could integrate the two : https://github.com/cozy/cozy-libs/issues/1165, thanks :)